db/migrate/20181108115009_create_user_associated_accounts.rb

```diff
@@ -0,0 +1,18 @@
+class CreateUserAssociatedAccounts < ActiveRecord::Migration[5.2]
+  def change
+    create_table :user_associated_accounts do |t|
+      t.string :provider_name, null: false
+      t.string :provider_uid, null: false
+      t.integer :user_id, null: false
+      t.datetime :last_used, null: false, default: -> { "CURRENT_TIMESTAMP" }
+      t.jsonb :info, null: false, default: {}
```

Comment:
I guess we are embracing our inner mongo ... 

mini_sql supports this which is nice: 

```
[15] pry(main)> DB.query("select '{\"bar\" : 1}'::json").first.json["bar"]
=> 1
```

Reply:
I am torn on using the JSON fields. On the one hand, it has no enforced schema, but on the other hand, each auth provider will be slightly different and this gives us the most flexibility.

I am pretty sure we will never want to do a "lookup" based on any of the info in the json fields, so I think it should be ok performance-wise. Activerecord also has pretty decent support for getting/setting the values.